timesketch/ui/static/components/explore/explore-event-directive.js

```diff
@@ -0,0 +1,138 @@
+
+                if ($scope.event._source.label.indexOf('__ts_comment') > -1) {
+                    $scope.comment = true;
+                    $scope.event._source.label.splice($scope.event._source.label.indexOf('__ts_comment'), 1)
+                }
+
+                $scope.toggleStar = function() {
+                    timesketchApi.saveEventAnnotation(
+                        $scope.sketchId,
+                        'label',
+                        '__ts_star',
+                        $scope.event._index,
+                        $scope.event._id,
+                        $scope.event._type).success(function(data) {})
+                };
+                $scope.getDetail = function() {
+                    if ($scope.eventdetail) {return}
+                    timesketchApi.getEvent(
+                        $scope.sketchId,
+                        $scope.event._index,
+                        $scope.event._id).success(function(data) {
+                            $scope.eventdetail = data.objects;
+                            $scope.comments = data.meta.comments;
+                        })
+                };
+                $scope.postComment = function() {
+                    timesketchApi.saveEventAnnotation(
+                        $scope.sketchId,
+                        'comment',
+                        $scope.formData.comment,
+                        $scope.event._index,
+                        $scope.event._id,
+                        $scope.event._type).success(function(data) {
+                            $scope.formData.comment = '';
+                            $scope.commentForm.$setPristine();
+                            $scope.comments.push(data['objects'][0]);
+                            $scope.comment = true;
+                        })
+                };
+            }
+        }
+    });
+
+    module.directive('tsTimelineColor', function () {
+        /**
+         * Render color background for timeline.
+         * @param timeline-colors - Object mapping index name to HEX color.
+         * @param index-id - Index name for the timeline.
+         */
+        return {
+            restrict: 'A',
+            scope: {
+                timelineColors: '=',
+                indexId: '='
+            },
+            link: function (scope, elem, attrs) {
+                elem.css("background", "#" + scope.timelineColors[scope.indexId])
+            }
+        }
+    });
+
+    module.directive('tsTimelineName', function () {
+        /**
+         * Render name for timeline.
+         * @param timeline-names - Object mapping index name to timeline name.
+         * @param index-id - Index id (in the datastore) for the timeline.
+         */
+        return {
+            restrict: 'E',
+            scope: {
+                timelineNames: '=',
+                indexId: '='
+            },
+            link: function (scope, elem, attrs) {
+                elem.text(scope.timelineNames[scope.indexId])
```

Comment:
True, I have moved this to the template instead and removed the whole directive.